### PR TITLE
[24] As a user, I can make an API call to search for keywords across my reports

### LINKED ISSUE #24: As a user, I can make an API call to search for keywords across my reports
## Why

Similar to the searching feature on the dashboard

## Acceptance Criteria

1. The user is able to make an API call to search for keywords

### PR DESCRIPTION
**Pull request by bterone**
Closes #24 

## What happened

✅: Search through user keywords using filter params
 
## Insight

Just reusing implementation from #58, without the `SearchHelper` module to parse the params.
 
## Proof Of Work

Postman Documentation - https://documenter.getpostman.com/view/16211750/TzeZEmGw

#### Success response
Screen Shot 2021-07-04 at 10 36 45 PM

#### Using filter
Screen Shot 2021-07-04 at 10 36 59 PM

#### Error response
Screen Shot 2021-07-04 at 10 37 37 PM

### CODE REVIEW

**Review by andyduong1920 — APPROVED** [review]

File: lib/g_searcher_web/controllers/api/search_result_controller.ex [hunk 1]
```diff
@@ -3,8 +3,9 @@ defmodule GSearcherWeb.API.SearchResultController do
 
   alias GSearcher.{SearchResults, SearchResultURLs}
   alias GSearcherWeb.ErrorHandler
+  alias GSearcherWeb.Validators.{ParamsValidator, SearchResultParams}
 
-  def index(conn, _params) do
+  def index(conn, params) when params == %{} do
```
Comment (hanam1ni) [inline lib/g_searcher_web/controllers/api/search_result_controller.ex:8]: How about?
```suggestion
  def index(conn, %{}) do
```

File: test/g_searcher_web/controllers/api/search_result_controller_test.exs [hunk 2]
```diff
@@ -60,6 +93,34 @@ defmodule GSearcherWeb.API.SearchResultControllerTest do
 
       assert json_response(conn, 200) == %{"data" => []}
     end
+
+    test "renders search results belonging to user given invalid params", %{conn: conn} do
```
Comment (hanam1ni) [inline test/g_searcher_web/controllers/api/search_result_controller_test.exs]: Should we update the test description here?
```suggestion
    test "renders 400 error response given invalid params", %{conn: conn} do
```

File: lib/g_searcher_web/controllers/api/search_result_controller.ex (hunk 1 above)
Comment (bterone) [reply lib/g_searcher_web/controllers/api/search_result_controller.ex:8]: Putting a blank map there is actually a valid pattern-match even if there are params 😄 
Screen Shot 2021-07-06 at 10 58 05 AM

We're kind of forced to do it this way, or maybe something like `map_size(params) == 0` 🤔

File: test/g_searcher_web/controllers/api/search_result_controller_test.exs (hunk 2 above)
Comment (bterone) [reply test/g_searcher_web/controllers/api/search_result_controller_test.exs]: Nice catch 😄 , I've updated the test description 9f4824ba7359218a1d49bd61f6247a36d8c2c045. 🕺

**Review by hanam1ni — APPROVED** [review]

**Pull request merged**